src/controllers/userController.js

```diff
@@ -140,6 +140,44 @@ export const editUser = async (req, res) => {
   }
 };
 
+export const changePassword = async (req, res) => {
+  const userId = req.userId;
+  const { currentPassword, newPassword } = req.body;
+
+  try {
+    const user = await User.findById(userId);
+
+    if (!(await user.comparePassword(currentPassword))) {
+      logger.error('Invalid credentials', {
+        method: req.method,
+        url: req.originalUrl,
+        userId: userId,
+      });
+      return res.status(400).json({ message: 'Invalid credentials' });
+    }
+
+    user.password = newPassword;
+    await user.save();
+
+    deleteTokensByUserId(userId, req.cookies.token);
```

Comment:
If the token is deleted, you are not supposed to be able to perform operations that require being logged in, right?

I have seen that I can execute several consecutive password changes without having to log in again.

Reply:
Yes that is correct. The token does not expire for the person who calls the method. You can try opening two sessions and you will see the other session is expired. The second parameter of the method is to exclude the token